src/functional/fencil_processors/source_modules/source_modules.py

```diff
@@ -15,11 +15,37 @@
 
 
 from dataclasses import dataclass
-from typing import Sequence
+from typing import Generic, Protocol, TypeVar
 
 import numpy
 
 
+class SupportedLanguage(Protocol):
+    """Ensures consistent code formatting along the pipeline."""
+
+    @property
+    def name(self) -> str:
+        ...
```

Comment:
It is intended to label different sets of language preferences (instances). If it was a class var it would be redundant (The class already has a readable name) and you would only be able to tell one set of `C++` preferences from another by looking at each of the values.